app/src/lib/editors/win32.ts

```diff
@@ -465,6 +493,31 @@ async function findApplication(editor: WindowsExternalEditor) {
     }
   }
 
+  if (editor.toolboxShellScriptName) {
+    const toolboxRegistryReference = [
+      CurrentUserUninstallKey('toolbox'),
+      Wow64LocalMachineUninstallKey('toolbox'),
+    ]
+
+    for (const { key, subKey } of toolboxRegistryReference) {
+      const keys = enumerateValues(key, subKey)
+
+      if (keys.length > 0) {
+        const editorPathInToolbox = Path.join(
+          getKeyOrEmpty(keys, 'UninstallString'),
+          '..',
+          '..',
+          'scripts',
+          `${editor.toolboxShellScriptName}.cmd`
+        )
+        const exists = await pathExists(editorPathInToolbox)
+        if (exists) {
+          return editorPathInToolbox
+        }
+      }
+    }
+  }
+
   return null
 }
```

Comment:
I think I'd extract this to a function:
```suggestion
  return findJetBrainsToolboxApplication(editor)
}

async function findJetBrainsToolboxApplication(editor: WindowsExternalEditor) {
  if (!editor.toolboxShellScriptName) {
    return null
  }
  
  const toolboxRegistryReference = [
    CurrentUserUninstallKey('toolbox'),
    Wow64LocalMachineUninstallKey('toolbox'),
  ]

  for (const { key, subKey } of toolboxRegistryReference) {
    const keys = enumerateValues(key, subKey)

    if (keys.length > 0) {
      const editorPathInToolbox = Path.join(
        getKeyOrEmpty(keys, 'UninstallString'),
        '..',
        '..',
        'scripts',
        `${editor.toolboxShellScriptName}.cmd`
      )
      const exists = await pathExists(editorPathInToolbox)
      if (exists) {
        return editorPathInToolbox
      }
    }
  }

  return null
}
```